client/platform/desktop/frontend/components/Settings.vue

```diff
@@ -20,10 +21,12 @@ export default defineComponent({
   setup() {
     // null values indicate initialization has not completed
     const smi = ref(null as NvidiaSmiReply | null);
-    const localSettings = settings;
+    // create a local copy of the global settings
+    const localSettings = cloneDeep(settings);
     const { arch, platform, version } = process;
-    const settingsAreValid = ref(true as boolean | string);
+    const settingsAreValid = ref(false as boolean | string);
     const gitHash = process.env.VUE_APP_GIT_HASH;
+    const readonlyMode = computed(() => settings.value.readonlyMode);
 
     onBeforeMount(async () => {
       settingsAreValid.value = await validateSettings(localSettings.value);
```

Comment:
Read the other comments first indicating that this may have `overrides: {}` even when an environment variable exists.
If the environment variable isn't a valid location should we still have this locked to prevent the user from manually changing it or should we indicate that they should adjust their environment variable to a valid location?